ccmlib/scylla_repository.py

```diff
@@ -303,6 +311,32 @@ def setup(version, verbose=True, skip_downloads=False):
     if skip_downloads:
         return directory_name(version), packages
 
+    if validate_by_hash and packages:
+        """
+        Validate if packages hash was changed and the new package(s) have to be downloaded 
+        """
+        map_field_to_dir_name = {"scylla_unified_package": CORE_PACKAGE_DIR_NAME,
+                                 "scylla_package": CORE_PACKAGE_DIR_NAME,
+                                 "scylla_tools_package": "scylla-tools-java",
+                                 "scylla_jmx_package": "scylla-jmx"
+                                 }
+        for package in zip(packages._fields, packages):
+            if not package[1]:
+                continue
+
+            new_hash = get_url_hash(package[1])
+            package_dir = map_field_to_dir_name[package[0]]
+            current_hash = get_installed_scylla_package_hash(source_file=Path(version_dir) / package_dir / SOURCE_FILE_NAME)
+            # Current hash may be None. It may be due to uncompleted downloading and installation.
+            # Or because of it is old installation that was not saved the hash yet.
+            # In any case the new download and installation should be performed.
+            # For this goal we need to remove existing folder and start downloading again
+            if not (new_hash and current_hash) or new_hash != current_hash:
+                # remove version_dir
```

Comment:
IIUC, `new_hash` should always be a non-empty string, right? so, probably we can simplify this a little bit. how about:

```py
for package in zip(packages._fields, packages):
    # ...
    if current_hash is None or new_hash != current_hash:
        rmdirs(version_dir)
        version_dir = None
        break
```

Reply:
i see. the `!` and `not` in the condition expression makes my eyes hurt. hence this suggestion. if we can truest the 'ETag' returned by S3, then it should not be empty.

Reply:
if `get_url_hash` always returns string, there no need to check for emptiness of it return value.

also `get_installed_scylla_package_hash` doesn't return None (at least not now), it always return a string, could be empty, so using `is None` wouldn't work as one might expect.

so `current_hash and new_hash != current_hash` should be suffice
if `get_url_hash` might return None
so `current_hash and new_hash and new_hash != current_hash` should do the trick.

Reply:
I am not sure if we can trust. 'ETag' is not mandatory. https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html

@fruch please, let you opinion

Reply:
@tchaikov I changed it, let me know if it is more clear